lock-api/src/test/java/com/palantir/lock/watch/ConjureLockRequestMetadataUtilsTest.java

```diff
@@ -0,0 +1,170 @@
+            CHANGE_METADATA_LIST.get(2),
+            LOCK_4,
+            CHANGE_METADATA_LIST.get(3));
+    private static final LockRequestMetadata LOCK_REQUEST_METADATA = LockRequestMetadata.of(LOCKS_WITH_METADATA);
+    private static final Map<Integer, ConjureChangeMetadata> CONJURE_LOCKS_WITH_METADATA = ImmutableMap.of(
+            0,
+            ConjureChangeMetadata.unchanged(ConjureUnchangedChangeMetadata.of()),
+            1,
+            ConjureChangeMetadata.updated(
+                    ConjureUpdatedChangeMetadata.of(Bytes.from(BYTES_OLD), Bytes.from(BYTES_NEW))),
+            2,
+            ConjureChangeMetadata.deleted(ConjureDeletedChangeMetadata.of(Bytes.from(BYTES_DELETED))),
+            3,
+            ConjureChangeMetadata.created(ConjureCreatedChangeMetadata.of(Bytes.from(BYTES_CREATED))));
+
+    private static ConjureMetadataConversionResult conjureMetadataConversionResult;
+
+    // This is a good candidate for a static construction method, but we would rather avoid testing internals of
+    // IndexEncodingUtils (checksum computation) within the tests for Conjure conversion.
+    @BeforeClass
+    public static void setup() {
+        KeyListChecksum checksum =
+                IndexEncodingUtils.computeChecksum(ConjureLockRequestMetadataUtils.DEFAULT_CHECKSUM_TYPE, LOCK_LIST);
+        ConjureLockRequestMetadata conjureMetadata = ConjureLockRequestMetadata.builder()
+                .indexToChangeMetadata(CONJURE_LOCKS_WITH_METADATA)
+                .lockListChecksum(
+                        ConjureLockDescriptorListChecksum.of(checksum.type().getId(), Bytes.from(checksum.value())))
+                .build();
+        conjureMetadataConversionResult = ConjureMetadataConversionResult.builder()
+                .lockList(LOCK_LIST)
+                .conjureMetadata(conjureMetadata)
+                .build();
+    }
+
+    @Test
+    public void convertsToConjureCorrectly() {
+        // ImmutableSet remembers insertion order, which is critical here
+        assertThat(ConjureLockRequestMetadataUtils.toConjureIndexEncoded(
+                        ImmutableSet.copyOf(LOCK_LIST), LOCK_REQUEST_METADATA))
+                .isEqualTo(conjureMetadataConversionResult);
+    }
+
+    @Test
+    public void convertsFromConjureCorrectly() {
+        assertThat(ConjureLockRequestMetadataUtils.fromConjureIndexEncoded(conjureMetadataConversionResult))
+                .isEqualTo(LOCK_REQUEST_METADATA);
+    }
+
+    @Test
+    public void convertingToAndFromConjureYieldsOriginalForRandomData() {
+        int randSeed = (int) System.currentTimeMillis();
+        Random random = new Random(randSeed);
+        Set<LockDescriptor> lockDescriptors = Stream.generate(UUID::randomUUID)
+                .map(UUID::toString)
+                .map(StringLockDescriptor::of)
+                .limit(1000)
+                .collect(Collectors.toSet());
+        List<ChangeMetadata> shuffled = new ArrayList<>(CHANGE_METADATA_LIST);
```

Comment:
Similar to the above, what are we trying to achieve with the random data generation? As what's good with this test now, you certainly have all change metadata types covered, but what if our checksum passes, but our metadata had swapped places?

Reply:
Ok so what I actually mainly want to test is if we handle sparse locks correctly (the case where not every lock has metadata associated with it. Random test is not actually necessary, I will just create some random locks as part of the setup.